### ISSUE TITLE
can not use image centos 7.4.1708

### ISSUE THREAD

**Issue by sriwiwat** [cloudbase/v-magine#21]
can not use image centos 7.4.1708

**Comment by morequ** [comment]
It seems that mirror list is outdated:
```
[    7.364071] dracut-initqueue[637]: Warning: Downloading 'http://centos.eecs.wsu.edu/7.4.1708/os/x86_64//LiveOS/squashfs.img' failed! 
[    7.370362] dracut-initqueue[637]: Warning: anaconda: failed to fetch stage2 from http://centos.eecs.wsu.edu/7.4.1708/os/x86_64/     
[    7.494022] random: crng init done                                                                                                   
[    8.261360] dracut-initqueue[637]: Warning: can't find installer mainimage path in .treeinfo                                         
[    8.282655] dracut-initqueue[637]: % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current                     
[    8.289779] dracut-initqueue[637]: Dload  Upload   Total   Spent    Left  Speed                                                      
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0  0 --:--:-- --:--:-- --:--:--     0                      
[    8.852629] dracut-initqueue[637]: curl: (22) The requested URL returned error: 404 Not Found
```

**Comment by maxgedroom** [comment]
There's a problem using "/7.4.1708/" at the URL. A readme file tells to use only "/7/", but i don't know how to change it prior the PXE boot...

This URL doesn't works: http://mirror.ufscar.br/centos/7.4.1708/os/x86_64/LiveOS/squashfs.img
But if you use this one, it will work: http://mirror.ufscar.br/centos/7/os/x86_64/LiveOS/squashfs.img

Any idea about the mirror list?

**Comment by MrFlutters** [comment]
The Mirror seems to be hard coded, [here](https://github.com/cloudbase/v-magine/blob/master/v_magine/centos.py)

This is no longer is no longer available with the method above and http://mirror.centos.org/centos/7.4.1708/ returns the readme mentioned above 

The only solutions i can think of would be to update the centos.py with the latest version of centos using http://mirrorlist.centos.org/?release=7&arch=x86_64&repo=os
**Or**
changing the default to point to http://vault.centos.org/7.4.1708/os/x86_64/ as this is now the only avalable mirror of 7.4.1708

**Comment by vivek-chandra** [comment]
Hello MrFlutters,

I edited the CentOS.py file, but I am not able to build exe file now. I am new to python programming, I downloaded the V-magine source code and edited the CentOS.py, later I installed PyIntaller but now I am not sure which file has to be passed as a parameter to the PyIntaller to get the exe.

Any suggestion would be highly appreciated. 

Thanks in advance
Vivek Chandra

**Comment by kimnamyeon** [comment]
on this subject
Is there any other way?

**Comment by hbonath** [comment]
+1 for me on  this too, I would like to try to install this, and am running into the same issue.
I'm going to attempt to load up an instance of HAProxy or NGINX on my LAN to answer and redirect/reverse-proxy to vault.centos.org to see if I can get around this for now, as I'm not really a coder and don't know the first thing about python.

**Comment by hbonath** [comment]
> +1 for me on this too, I would like to try to install this, and am running into the same issue.
> I'm going to attempt to load up an instance of HAProxy or NGINX on my LAN to answer and redirect/reverse-proxy to vault.centos.org to see if I can get around this for now, as I'm not really a coder and don't know the first thing about python.

So this seems to have worked, I set up a VM running OpenBSD and HAProxy, pointed the dns for "mirror.centos.org" to the IP of the OpenBSD box and got v-magine to install. 
Here's my super basic HAProxy config, the backend IP is for "vault.centos.org"

`global
        log 127.0.0.1   local0 debug
        maxconn 1024
        chroot /var/haproxy
        uid 604
        gid 604
        daemon
        pidfile /var/run/haproxy.pid

defaults
        log     global
        mode    http
        option  httplog
        option  dontlognull
        option  redispatch
        retries 3
        maxconn 2000

frontend haproxy
        bind *:80
        default_backend httpd

backend httpd
        option forwardfor
        server www 162.251.108.7:80 check
`

**Comment by adcura** [comment]
I had the same issue, seem to have resolved using a local DNS server that the deployment was referring to, created a zone and host for one of the mirrors : centos.mirror.ausnetservers.net.au because it began with the centos zone name, I was able to redirect it to the ip of vault.centos.org: 162.251.108.7 and the deployment seems to be proceeding past this point where it had failed before - dodgy workaround, but wasn't able to recompile the v-magine package from source with an updated centos.py version no.

Anyone in a position to recompile?  Could we get a test done using a newer/supported release of Centos?  @alexpilotti ?

**Comment by jmsj05** [comment]
Has there been any update on a fix?

**Comment by alexpilotti** [comment]
Yes, the issue is due to CentOS mirrors stop serving older versions shortly after a new one comes up. The repo has been updated to 7.5. We're working on a new release ATM.

**Comment by alexpilotti** [comment]
Duplicate of #20

**Comment by alexpilotti** [comment]
As a side note the reason why the CentOS version is fixed is that the linux PXE resources work only with that specific version, so every time we need to update both. We can put the version in a config file, so at least it can be changed without recompiling (the CentOS PXE bits can be copied from the latest CentOS minimal iso)

**Comment by alexpilotti** [comment]
Fixed by 3e50be8a619f772711308199b4cad95c6d7f7f92
